compiler/code-gen/vertex-compiler.cpp

```diff
@@ -1178,7 +1185,8 @@ void compile_switch_int(VertexAdaptor<op_switch> root, CodeGenerator &W) {
       if (val->type() == op_int_const) {
         const std::string &str = val.as<op_int_const>()->str_val;
         W << str;
-        kphp_error(used.insert(str).second, fmt_format("Switch: repeated cases found [{}]", str));
+        const std::string op = root->is_match ? "Match" : "Switch";
+        kphp_error(used.insert(str).second, fmt_format("{}: repeated cases found [{}]", op, str));
```

Comment:
Here you evaluate `op` (construct std::string) every time you enter `if` branch, and always (without an error) it's immediately destroyed. It's better to inline this ternary as argument into `kphp_error` call